test/nodes/test_reader.py

```diff
@@ -10,12 +10,19 @@
 from haystack.nodes.reader.farm import FARMReader
 
 
+@pytest.fixture
+def prediction(reader, docs):
+    prediction = reader.predict(query="Who lives in Berlin?", documents=docs, top_k=5)
+    return prediction
+
+
 def test_reader_basic(reader):
     assert reader is not None
     assert isinstance(reader, BaseReader)
 
 
-def test_output(prediction):
+def test_output(reader, docs):
```

Comment:
Could you please explain why we shouldn't use the `prediction` fixture here?

Reply:
Honestly it's because it made waaay simpler to debug the tests :smile: There's no good way to step into a fixture initialization when debugging a test, so putting the call into the test itself was necessary for me to see what was going on. I thought about removing it, but I don't see any benefit of doing so, so I eventually left it where it is.

Do you see any benefit of using a fixture here? I really fail to see any. It just makes the tests harder to debug :sweat_smile:

Reply:
The main reason that I see is consistency with the other tests in `reader.py` that use the fixture too. Another small reason could be speed because we could reuse the same fixture without running the `predict()` method multiple times. That could work if we set the scope of the fixture to module (instead of the default scope function).